postgresql-plugin/src/e2e-test/java/io/cdap/plugin/PostgresqlClient.java

```diff
@@ -0,0 +1,158 @@
+    throws SQLException, ClassNotFoundException {
+    String getSourceQuery = "SELECT * FROM " + schema + "." + sourceTable;
+    String getTargetQuery = "SELECT * FROM " + schema + "." + targetTable;
+    try (Connection connect = getPostgresqlConnection()) {
+      connect.setHoldability(ResultSet.HOLD_CURSORS_OVER_COMMIT);
+      Statement statement1 = connect.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE,
+                                                     ResultSet.HOLD_CURSORS_OVER_COMMIT);
+      Statement statement2 = connect.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE,
+                                                     ResultSet.HOLD_CURSORS_OVER_COMMIT);
+      ResultSet rsSource = statement1.executeQuery(getSourceQuery);
+      ResultSet rsTarget = statement2.executeQuery(getTargetQuery);
+      return compareResultSetData(rsSource, rsTarget);
+    }
+  }
+
+  /**
+   * Compares the result Set data in source table and sink table..
+   * @param rsSource result set of the source table data
+   * @param rsTarget result set of the target table data
+   * @return true if rsSource matches rsTarget
+   */
+  public static boolean compareResultSetData(ResultSet rsSource, ResultSet rsTarget) throws SQLException {
+    ResultSetMetaData mdSource = rsSource.getMetaData();
+    ResultSetMetaData mdTarget = rsTarget.getMetaData();
+    int columnCountSource = mdSource.getColumnCount();
+    int columnCountTarget = mdTarget.getColumnCount();
+    Assert.assertEquals("Number of columns in source and target are not equal",
+                        columnCountSource, columnCountTarget);
+    while (rsSource.next() && rsTarget.next()) {
+      int currentColumnCount = 1;
+      while (currentColumnCount <= columnCountSource) {
+        String columnTypeName = mdSource.getColumnTypeName(currentColumnCount);
+        int columnType = mdSource.getColumnType(currentColumnCount);
+        String columnName = mdSource.getColumnName(currentColumnCount);
+        if (columnType == Types.TIMESTAMP) {
+          GregorianCalendar gc = new GregorianCalendar(TimeZone.getTimeZone("UTC"));
+          gc.setGregorianChange(new Date(Long.MIN_VALUE));
+          Timestamp sourceTS = rsSource.getTimestamp(currentColumnCount, gc);
+          Timestamp targetTS = rsTarget.getTimestamp(currentColumnCount, gc);
+          Assert.assertTrue(String.format("Different values found for column : %s", columnName),
+                            sourceTS.equals(targetTS));
+        } else {
+          String sourceString = rsSource.getString(currentColumnCount);
+          String targetString = rsTarget.getString(currentColumnCount);
+          Assert.assertTrue(String.format("Different values found for column : %s", columnName),
+                            String.valueOf(sourceString).equals(String.valueOf(targetString)));
```

Comment:
We should not require doing String.valueOf(String) here.

Reply:
done